sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SQLConfHelper.scala

```diff
@@ -29,4 +30,32 @@ trait SQLConfHelper {
    * See [[SQLConf.get]] for more information.
    */
   def conf: SQLConf = SQLConf.get
+
+  /**
+   * Sets all SQL configurations specified in `pairs`, calls `f`, and then restores all SQL
+   * configurations.
+   */
+  protected def withSQLConf[T](pairs: (String, String)*)(f: => T): T = {
```

Comment:
Oh, this looks like an independent contribution too. Sorry for asking this again, but could you spin-off this too?

The new PR title could be the following, @ulysses-you .
```
Move `withSQLConf` from SQLHelper trait to `SQLConfHelper` trait
```